### PR TITLE
Added authentication check for Datasetsubmission PUT API

### PR DESCRIPTION
**Pull request by praneethpr**

### CODE REVIEW

File: src/Controller/Api/DatasetSubmissionController.php [hunk 1]
```diff
@@ -182,6 +183,9 @@ public function postAction(Request $request)
      *     )
      * )
      *
+     * @IsGranted("IS_AUTHENTICATED_FULLY")
+     * @IsGranted("CAN_EDIT", subject="datasetSubmission")
```
Comment (mickel1138) [inline src/Controller/Api/DatasetSubmissionController.php:187]: I'm not sure if this `subject` is available, you may have to do it with that magic conversion method?

Comment (mickel1138) [reply src/Controller/Api/DatasetSubmissionController.php:187]: Actually, don't need this line at all, entityHandler checks that.

**Review by mickel1138 — APPROVED** [review]
This works always now.

**Review by fightingtexasaggie — APPROVED** [review]
works, athough there isn't a re-direct when clicking between tabs, just the noty, but the message is reasonable.

**Pull request merged**